pom.xml

```diff
@@ -46,6 +76,23 @@
           </descriptorRefs>
         </configuration>
       </plugin>
+
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-compiler-plugin</artifactId>
+        <version>3.3</version>
+        <configuration>
+          <compilerArgument>
+            -Xlint:unchecked,deprecation
+          </compilerArgument>
+        </configuration>
+      </plugin>
+
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-javadoc-plugin</artifactId>
```

Comment:
IIRC in such configuration the plugin runs Javadoc generation only during the release or explicit `javadoc:javadoc` call. Maybe not, but it worth to make sure to prevent last-minute failure during the release